realm/src/main/java/io/realm/BaseRealm.java

```diff
@@ -77,6 +88,7 @@
     protected SharedGroupManager sharedGroupManager;
     protected boolean autoRefresh;
     Handler handler;
+    ColumnIndices columnIndices;
```

Comment:
It would yes, but right now RealmQuery, RealmList and RealmResults only know about `BaseRealm`. I have a feeling that the right answer is probably letting the Realm subclass convert all Class references to Strings right away, but it has of yet unknown performance/code clarity implications.

I would prefer to address this in another PR.